docs/src/lib/types.md

```diff
@@ -16,7 +16,36 @@ It is not intended as a fully generic interface for working with tabular data, w
 interfaces defined by [Tables.jl](https://github.com/JuliaData/Tables.jl/) instead.
 
 `DataFrame` is the most fundamental subtype of `AbstractDataFrame`, which stores a set of columns
-as `AbstractVector` objects.
+as `AbstractVector` objects. When a `DataFrame` is constructed columns *are not* copied, if possible.
+The exception are `DataFrame(::DataFrame)` and `DataFrame(::SubDataFrame)` constructors
+which perform a copy of the columns.
+
+From the moment of its construction `DataFrame` assumes *ownership* of its columns. This means that
+methods in DataFrames.jl package assume that columns of the data frame *are not* mutated by
+methods other than provided by the DataFrames.jl package that accept this data frame as an argument.
```

Comment:
This isn't very clear IMHO.

Reply:
OK - I have rewritten it. Feel free to correct inline 😄, as this is a hard passage to write.

Reply:
Actually I'm not sure what you want to say exactly in this sentence. :-)

To me, what "ownership" means here is that:
- resizing column vectors will corrupt the `DataFrame`
- mutating their contents will affect the `DataFrame`, and in particular reordering observations in only one column will make observations (rows) inconsistent and corrupt a child `SubDataFrame`
- modifying grouping column vectors will corrupt a child `GroupedDataFrame`

Additionally, some functions ending with a `!` mutate the column vectors. other are guaranteed not to mutate them.

So I would tend to move the last part with the list here, add the second bullet, and a mention about functions like `sort!` and `dropmissing!`.

Reply:
I have tried another approach of the description

Reply:
I've made a proposal. I'm starting to think this should have its own section, as it completely breaks the logic of the presentation of types. Maybe we could put this in the user guide rather than in the API: that's not too technical/repetitive and that's really important for users.

Reply:
OK - I will move it and we will see how it goes.